engine/class_modules/sc_death_knight.cpp

```diff
@@ -8907,11 +8908,23 @@ void death_knight_t::default_apl_blood()
   // Cooldowns
   def -> add_action( this, "Vampiric Blood" );
   def -> add_action( "potion,if=buff.dancing_rune_weapon.up" );
-  def -> add_action( this, "Dancing Rune Weapon", "if=!talent.blooddrinker.enabled|!cooldown.blooddrinker.ready" );
-  def -> add_talent( this, "Tombstone", "if=buff.bone_shield.stack>=7" );
+  //def -> add_action( "call_action_list,name=covenants" );
   def -> add_action( "call_action_list,name=essences" );
   def -> add_action( "call_action_list,name=standard" );
-
+  /*
+  // Hardcast Death's Due with no CS proc if there is < 3s on the buff.
+  covenants -> add_action( "deaths_due,if=covenant.night_fae&(!buff.deaths_due.up|buff.deaths_due.remains<5|buff.crimson_scourge.up)" );
+  // Cast HS on the last tick of Death's Due to refresh the buff as late as possible
+  covenants -> add_action( "heart_strike,if=covenant.night_fae&death_and_decay.remains<3" );
+  // If we're venthyr and we're too high on RP as swarming mist is about to come up, dump some
+  covenants -> add_action( "death_strike,if=covenant.venthyr&runic_power>70&cooldown.swarming_mist.remains<3" );
+  // Cast swarming mist if within 3s of DRW
+  covenants -> add_action( "swarming_mist,if=covenant.venthyr&(dancing_rune_weapon.cooldown.remains<3|!buff.dancing_rune_weapon.up)" );
+  // Cast abomination limb if available and DRW isn't (to cast it prior to DRW), or DRW is not up.
+  covenants -> add_action( "abomination_limb,if=covenant.necrolord&(dancing_rune_weapon.cooldown.remains<3|!buff.dancing_rune_weapon.up)" );
+  // Finally, we cast shackle if available and DRW isn't (to cast it prior to DRW), or DRW is not up. TODO: pool 2 runes prior, modify APL to prioritize spreading on multi-target
+  covenants -> add_action( "shackle_the_unworthy,if=covenant.kyrian&(dancing_rune_weapon.cooldown.remains<3|!buff.dancing_rune_weapon.up)" );
```

Comment:
It's probably better to add comments inside add_action() than outside of it if you want them to show up in the generated profile